src/pip/_internal/utils/filesystem.py

```diff
@@ -13,19 +13,22 @@ def check_path_owner(path):
 
     previous = None
     while path != previous:
-        if not os.path.lexists(path):
-            previous, path = path, os.path.dirname(path)
-        else:
-            # Check if path is not writable by current user.
-            if os.geteuid() != 0:
-                return os.access(path, os.W_OK)
+        if os.path.lexists(path):
+            break
+        previous, path = path, os.path.dirname(path)
 
-            # Special handling for root user in order to handle properly
-            # cases where users use sudo without -H flag.
-            try:
-                path_uid = get_path_uid(path)
-            except OSError:
-                return False
-            else:
-                return path_uid == 0
-    return False  # assume we don't own the path
+    # Assume we don't own the root path, which would make the following True.
+    if path == previous:
+        return False
```

Comment:
These two lines can be simplified to--

```python
else:
    return False
```

This is because the while loop is exited normally precisely when `path == previous`.